tidysic/settings/formatted_string.py

```diff
@@ -0,0 +1,71 @@
+import re
+
+from tidysic.file.taggable import Taggable
+
+
+class FormattedString:
+    def __init__(self, raw_string: str):
+        try:
+            self.validate(raw_string)
+        except ValueError as e:
+            raise ValueError(f"Could not create formatted string, {e}")
+
+        self.raw_string = raw_string
+
+    def write(self, taggable: Taggable):
+        """
+        Produces the string built using the tags found in the given taggable
+        """
+        pattern = r"\{(.*?\{(\w+)(:.+?)?\}.*?)\}"
+        matches = re.findall(pattern, self.raw_string)
+
+        return_string = self.raw_string
+
+        substitutions = []
+        for to_substitute, tag_name, format_spec in matches:
+
+            value = getattr(taggable, tag_name, None)
+            if tag_name in Taggable.get_numeric_tag_names() and value:
```

Comment:
I usually like to check first single values before iterating over a list of unknown size, i.e.: 

```python
if value and tag_name in Taggable.get_numeric_tag_names()
```

Reason being that if value is `False`, Python won't even iterate over the list. Obviously in this case the performance is not affected, but it is more like a "coding style" comment 😉

Reply:
Makes sense. I was going for a more "conversational" style of condition. "If it should be a number and we got something, then..." and "If it should be a string but there is none, say "unknown tag". I'll switch to your side since mine has less justification.